app/src/main/java/com/quran/labs/androidquran/SearchWidget.java

```diff
@@ -0,0 +1,67 @@
+package com.quran.labs.androidquran;
+
+import android.app.PendingIntent;
+import android.appwidget.AppWidgetManager;
+import android.appwidget.AppWidgetProvider;
+import android.content.Context;
+import android.content.Intent;
+import android.widget.RemoteViews;
+
+import com.quran.labs.androidquran.ui.PagerActivity;
+
+
+/**
+ * Implementation of App Widget functionality.
+ */
+public class SearchWidget extends AppWidgetProvider {
+
+  @Override
+  public void onUpdate(Context context, AppWidgetManager appWidgetManager, int[] appWidgetIds) {
+    // There may be multiple widgets active, so update all of them
+    for (int appWidgetId : appWidgetIds) {
```

Comment:
I am not sure about footprint on memory, but this for loop looks neater than the one used in `BookmarksWidget.onUpdate()`.